platform/view/services/db/driver/sql/common/pagination.go

```diff
@@ -85,25 +87,140 @@ func (p *OffsetPagination) GoBack(numOfpages int) (driver.Pagination, error) {
 func (p *OffsetPagination) Prev() (driver.Pagination, error) { return p.GoBack(1) }
 func (p *OffsetPagination) Next() (driver.Pagination, error) { return p.GoForward(1) }
 
+type KeysetPagination struct {
+	offset   int
+	pageSize int
+	// name of the field in the database that is a unique id of the records
+	sqlIdName string
+	// name of the field in the struct that is returned from the database
+	idFieldName string
+	// the first and last id values in the page
+	firstId string
+	lastId  string
+}
+
+func NewKeysetPagination(offset int, pageSize int, sqlIdName string, idFieldName string) (*KeysetPagination, error) {
+	if offset < 0 {
+		return nil, fmt.Errorf("offset shoud be grater than zero. Offset: %d", offset)
+	}
+	if pageSize < 0 {
+		return nil, fmt.Errorf("page size shoud be grater than zero. pageSize: %d", pageSize)
+	}
+	return &KeysetPagination{
+		offset:      offset,
+		pageSize:    pageSize,
+		sqlIdName:   sqlIdName,
+		idFieldName: idFieldName,
+		firstId:     "",
```

Comment:
you don't need to set empty strings. you can omit them and they default to the empty string